src/main/java/org/jenkinsci/plugins/gitlab/GitlabMergeRequestWrapper.java

```diff
@@ -349,6 +350,22 @@ public GitlabNote createNote(String message, boolean shouldClose, boolean should
 
     }
 
+    public GitlabCommitStatus createCommitStatus(String status, String targetUrl) {
+        try {
+            GitlabAPI api = builder.getGitlab().get();
+            GitlabMergeRequest mergeRequest = api.getMergeRequest(project, id);
+            GitlabCommit latestCommit = getLatestCommit(mergeRequest, api);
+
+            if (latestCommit != null) {
+                return api.createCommitStatus(project, latestCommit.getId(), status, mergeRequest.getSourceBranch(), "Jenkins", targetUrl, null);
```

Comment:
You're correct. If a new commit is pushed between a build being started and finishing, this will add the success to the incorrect commit.

@timols What would the best way to save the commit id that we started the build on be?